File: AdamClassLib/Sample Code/APAX/Win32/VB/APAX-Coupler-Sample/APAX_IO_Module_Library/IPForm.vb
Public Class IPForm
    Public Delegate Sub EventHandler_ApplyIPAddressClick(ByVal szVal As String)


    Public Event ApplyIPAddressClick As EventHandler_ApplyIPAddressClick

    Private Sub button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles button1.Click
        label1.Text = IPAddressText.Text
        RaiseEvent ApplyIPAddressClick(IPAddressText.Text)
        Me.Close()
    End Sub
End Class

File: AdamClassLib/Sample Code/APAX/WinCE/CSharp/APAX-Coupler-Sample/Apax_IO_Module_Library/IPForm.cs
using System;

using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace Apax_IO_Module_Library
{
    public partial class IPForm : Form
    {
        public delegate void EventHandler_ApplyIPAddressClick(string szVal);
        public event EventHandler_ApplyIPAddressClick ApplyIPAddressClick;
        public IPForm()
        {
            InitializeComponent();
        }

        private void button1_Click_1(object sender, EventArgs e)
        {
            if (ApplyIPAddressClick != null)
            {
                label1.Text = IPAddressText.Text;

                ApplyIPAddressClick(IPAddressText.Text);

            }
            this.Close();
        }
    }
}